pkg/controller/mattermostcluster/mattermostcluster_controller.go

```diff
@@ -100,54 +110,524 @@ func (r *ReconcileMattermostCluster) Reconcile(request reconcile.Request) (recon
+		if err != nil {
+			return err
+		}
+		_ = controllerutil.SetControllerReference(mattermost, foundDB, r.scheme)
+	}
+
+	return nil
+}
+
+func (r *ReconcileMattermostCluster) checkDBPostgresDeployment(mattermost *mattermostv1alpha1.MattermostCluster, reqLogger logr.Logger) error {
+	dbExist := false
+
+	//TODO: Create the logic to check if the DB Postgres already exist or changed otherwise create
+	if dbExist {
+		return r.client.Update(context.TODO(), mattermost)
+	}
+	return nil
+}
+
+func (r *ReconcileMattermostCluster) checkMinioDeployment(mattermost *mattermostv1alpha1.MattermostCluster, reqLogger logr.Logger) error {
+	minioExist := false
+
+	//TODO: Create the logic to check if the Minio already exist or changed otherwise create
+	if minioExist {
+		return r.client.Update(context.TODO(), mattermost)
+	}
+	return nil
+}
+
+func (r *ReconcileMattermostCluster) getSecrets(mattermost *mattermostv1alpha1.MattermostCluster, reqLogger logr.Logger) (string, error) {
+	dbSecretName := fmt.Sprintf("%s-db-root-password", mattermost.Name)
+	dbSecret := &corev1.Secret{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: dbSecretName, Namespace: mattermost.Namespace}, dbSecret)
+	if err != nil {
+		return "", err
+	}
+	return string(dbSecret.Data["password"]), nil
+}
+
+func (r *ReconcileMattermostCluster) checkMattermostDeployment(mattermost *mattermostv1alpha1.MattermostCluster, reqLogger logr.Logger) error {
+	//TODO: Create the logic to deploy MM including service and ingress
+	err := r.serviceForMM(mattermost, reqLogger)
+	if err != nil {
+		return err
+	}
+
+	err = r.ingressForMM(mattermost, reqLogger)
+	if err != nil {
+		return err
+	}
+
+	err = r.deploymentForMM(mattermost, reqLogger)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *ReconcileMattermostCluster) serviceForMM(mattermost *mattermostv1alpha1.MattermostCluster, reqLogger logr.Logger) error {
+	mattermostPort := corev1.ServicePort{Port: 8065}
+	svc := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels:    map[string]string{constants.ClusterLabel: mattermost.Name},
+			Name:      mattermost.Name,
+			Namespace: mattermost.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(mattermost, schema.GroupVersionKind{
+					Group:   mattermostv1alpha1.SchemeGroupVersion.Group,
+					Version: mattermostv1alpha1.SchemeGroupVersion.Version,
+					Kind:    "MattermostCluster",
+				}),
+			},
+			Annotations: map[string]string{
+				"service.alpha.kubernetes.io/tolerate-unready-endpoints": "true",
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{mattermostPort},
+			Selector: map[string]string{
+				constants.ClusterLabel: mattermost.Name,
+			},
+			ClusterIP: corev1.ClusterIPNone,
+		},
+	}
+
+	foundService := &corev1.Service{}
+	errGet := r.client.Get(context.TODO(), types.NamespacedName{Name: mattermost.Name, Namespace: mattermost.Namespace}, foundService)
```

Comment:
nit: Looks like if one already exists, we can bail out early. Perhaps invert this to check up front and do exactly that, and then you can outdent the code that creates one instead?

Similar feedback in the methods below.